core/controlplane/config/templates/cloud-config-controller

```diff
@@ -205,6 +205,7 @@ coreos:
         Wants=flanneld.service cfn-etcd-environment.service
         After=cfn-etcd-environment.service
         [Service]
+        EnvironmentFile=/etc/environment
```

Comment:
Is this probably an improvement irrelevant to the local dnsmasq thing?
If that's the case, would you mind submitting an another PR dedicated for it, or at least splitting commits?